### PR TITLE
minmax_element

### PR DESCRIPTION
**Pull request by fndrd1d1**

### CODE REVIEW

File: algorithms/minmax_element.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{
+    std::vector<int> v{1, 2, 3, 4, 3};
+    auto [min, max] = alg::minmax_element(begin(v), end(v), std::less());
+    auto [min2, max2] = std::minmax_element(begin(v), end(v), std::less());
+
+    EXPECT_EQ(*min, *min2);
+    EXPECT_EQ(*max, *max2);
+}
+
+int main(int argc, char **argv)
```
Comment (0xPwner) [inline algorithms/minmax_element.cpp:70]: I think we should have only one main that is located in project root which will execute all the tests in the future. 
So adding tests can be easy

File: algorithms/minmax_element.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            }
+            if (cmp(*potentialMin, *min))
+            {
+                min = potentialMin;
+            }
+            if (!cmp(*potentialMax, *max))
+            {
+                max = potentialMax;
+            }
+            ++first;
```
Comment (kafteji) [inline algorithms/minmax_element.cpp]: first += 2;

File: algorithms/minmax_element.cpp (hunk 1 above)
Comment (kafteji) [inline algorithms/minmax_element.cpp]: I think we should have only one main that is located in project root which will execute all the tests in the future.
So adding tests can be easy.

File: algorithms/minmax_element.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            {
+                min = first;
+            }
+            if (!cmp(*first, *max))
+            {
+                max = first;
+            }
+        }
+
+        return {min, max};
```
Comment (kafteji) [inline algorithms/minmax_element.cpp]: i would use std::make_pair(min, max)

File: algorithms/minmax_element.cpp [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+namespace alg
+{
+    // idea: go through container two elements at a time
+    template <typename I, typename Compare>
+    // requires I is a ForwardIterator
+    // Compare is StrictWeakOrdering on ValueType(I)
+    std::pair<I, I> minmax_element(I first, I last, Compare cmp)
+    {
+        if (first == last)
+            return {last, last};
```
Comment (kafteji) [inline algorithms/minmax_element.cpp]: std::make_pair(last, last)

**Review by kafteji — CHANGES_REQUESTED** [review]
nice manos

File: algorithms/minmax_element.cpp (hunk 2 above)
Comment (fndrd1d1) [reply algorithms/minmax_element.cpp]: ✅

File: algorithms/minmax_element.cpp (hunk 1 above)
Comment (fndrd1d1) [reply algorithms/minmax_element.cpp]: ✅

File: algorithms/minmax_element.cpp (hunk 3 above)
Comment (fndrd1d1) [reply algorithms/minmax_element.cpp]: ✅

File: algorithms/minmax_element.cpp (hunk 4 above)
Comment (fndrd1d1) [reply algorithms/minmax_element.cpp]: ✅

**Review by kafteji — APPROVED** [review]
LGTM

**Pull request merged**